solace_prometheus_exporter.go

```diff
@@ -224,10 +245,62 @@ func main() {
 
 	// start server
 	if conf.EnableTLS {
-		if err := http.ListenAndServeTLS(conf.ListenAddr, conf.Certificate, conf.PrivateKey, nil); err != nil {
+		var tlsCert tls.Certificate
```

Comment:
Please move the whole TLS logic to an separate file and method. Try to avoid bloat up single files.